UnityProject/Assets/Scripts/Util/SweetExtensions.cs

```diff
@@ -82,7 +82,11 @@ public static T GetRandom<T>(this List<T> list)
 
 	public static uint NetId(this GameObject go)
 	{
-		return go ? go.GetComponent<NetworkIdentity>().netId : global::NetId.Invalid; //maxValue is invalid (see NetId.cs)
+		return go == null
+			? global::NetId.Invalid
+			: go.GetComponent<Matrix>()
+				? go.GetComponentInChildren<NetworkIdentity>().netId 
```

Comment:
Splitting it out into If statements would be better like Gilles said and then do TryGetComponent<Matrix>(out var matrix)

and then returning matrix.NetworkedMatrix.MatrixSync.netId